### PR TITLE
[PRMP-1221] Lambda module - accept >10 iam_role_policies

### PR DESCRIPTION
**Pull request by abid-nhs**

### CODE REVIEW

File: infrastructure/modules/app_config/output.tf [hunk 1]
```diff
@@ -10,6 +10,10 @@ output "app_config_configuration_profile_id" {
   value = aws_appconfig_configuration_profile.ndr-app-config-profile.configuration_profile_id
 }
 
+output "app_config_policy" {
+  value = aws_iam_policy.app_config_policy.policy
```
Comment (chrisbloe-nhse) [inline infrastructure/modules/app_config/output.tf:14]: Probably needs to be a `data` block instead.

File: infrastructure/modules/lambda/main.tf [hunk 2]
```diff
@@ -60,10 +60,31 @@ resource "aws_iam_role" "lambda_execution_role" {
...
+data "aws_iam_policy_document" "merged_policy" {
+  source_policy_documents = var.iam_role_policy_documents
+}
+
+resource "aws_iam_policy" "lambda_combined_policy" {
+  name   = "${terraform.workspace}_${var.name}_combined_policy"
+  policy = data.aws_iam_policy_document.merged_policy.json
+}
+
+variable "default_policies" {
```
Comment (chrisbloe-nhse) [inline infrastructure/modules/lambda/main.tf:72]: This needs to be in `variable.tf`

Comment (abid-nhs) [reply infrastructure/modules/lambda/main.tf]: Done

File: infrastructure/modules/app_config/output.tf (hunk 1 above)
Comment (abid-nhs) [reply infrastructure/modules/app_config/output.tf]: Done

File: infrastructure/modules/lambda/main.tf [hunk 3]
```diff
@@ -60,10 +60,24 @@ resource "aws_iam_role" "lambda_execution_role" {
   assume_role_policy = data.aws_iam_policy_document.assume_role.json
 }
 
+data "aws_iam_policy_document" "merged_policy" {
+  source_policy_documents = var.iam_role_policy_documents
+}
+
+resource "aws_iam_policy" "lambda_combined_policy" {
```
Comment (chrisbloe) [inline infrastructure/modules/lambda/main.tf:67]: `"combined_policies"`

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 4]
```diff
@@ -23,6 +22,10 @@ module "manage-nrl-pointer-lambda" {
   }
   is_gateway_integration_needed = false
   is_invoked_from_gateway       = false
+
+  depends_on = [
```
Comment (chrisbloe) [inline infrastructure/lambda-manage-nrl-pointer.tf:26]: Implicit

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 5]
```diff
@@ -33,6 +36,7 @@ module "manage-nrl-pointer-alarm" {
   namespace            = "AWS/Lambda"
   alarm_actions        = [module.manage-nrl-pointer-alarm-topic.arn]
   ok_actions           = [module.manage-nrl-pointer-alarm-topic.arn]
+  depends_on           = [module.manage-nrl-pointer-lambda, module.manage-nrl-pointer-alarm-topic]
```
Comment (chrisbloe) [inline infrastructure/lambda-manage-nrl-pointer.tf:39]: Implicit

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 6]
```diff
@@ -62,6 +66,8 @@ module "manage-nrl-pointer-alarm-topic" {
       }
     ]
   })
+
+  depends_on = [module.manage-nrl-pointer-lambda, module.sns_encryption_key]
```
Comment (chrisbloe) [inline infrastructure/lambda-manage-nrl-pointer.tf:70]: Implicit

File: infrastructure/lambda-manage-nrl-pointer.tf [hunk 7]
```diff
@@ -77,6 +83,15 @@ resource "aws_lambda_event_source_mapping" "nrl_pointer_lambda" {
       })
     }
   }
+
+  scaling_config {
+    maximum_concurrency = local.bulk_upload_lambda_concurrent_limit
+  }
+
+  depends_on = [
```
Comment (chrisbloe) [inline infrastructure/lambda-manage-nrl-pointer.tf:91]: Implicit

File: infrastructure/modules/lambda/main.tf (hunk 3 above)
Comment (abid-nhs) [reply infrastructure/modules/lambda/main.tf]: done

File: infrastructure/lambda-manage-nrl-pointer.tf (hunk 4 above)
Comment (abid-nhs) [reply infrastructure/lambda-manage-nrl-pointer.tf:26]: This came through from PRMP-1122

File: infrastructure/lambda-manage-nrl-pointer.tf (hunk 5 above)
Comment (abid-nhs) [reply infrastructure/lambda-manage-nrl-pointer.tf:39]: This came through from PRMP-1122

File: infrastructure/lambda-manage-nrl-pointer.tf (hunk 6 above)
Comment (abid-nhs) [reply infrastructure/lambda-manage-nrl-pointer.tf:70]: This came through from PRMP-1122

File: infrastructure/lambda-manage-nrl-pointer.tf (hunk 7 above)
Comment (abid-nhs) [reply infrastructure/lambda-manage-nrl-pointer.tf:91]: This came through from PRMP-1122

**Pull request merged**